app/services/message_service.rb

```diff
@@ -0,0 +1,29 @@
+class MessageService
+  attr_accessor :text, :conversation, :user
+
+  def initialize(text, conversation, user)
+    @text = text
+    @conversation = conversation
+    @user = user
+  end
+
+  def create_message
+    @message = conversation.messages
+                           .create!(text: text, user_id: user.id)
+    notify
+    @message
+  end
+
+  def notify
+    return if target_match
+
+    users = conversation.users.reject { |each| each == user }
```

Comment:
The conversation has only 2 users, right? Why do you have has_many?
I would move this to the conversation model and try to use a query instead of reject (reject is in memory, the query is in the database directly)

Reply:
An example for the comment above:
In the conversation model:
```ruby
def other_parts(user)
  users.where.not(id: user.id)
end
```
(I think you have a scope to exclude a user, you can use that)

Than you call: `conversation.other_parts(user)`

The difference is that reject exclude the user in memory and the where query in the database (so better performance)

Reply:
It was done like this to open the possibility of multiple user conversations in the future. 
Also, in the beginning I tried to do something like first_user, second_user on conversation model but didn't end up good, I needed the through model to store some attributes like unread messages for each user.

Reply:
I couldn't come up with a query for this. Any ideas?

Reply:
I left a comment above. Did you try that?